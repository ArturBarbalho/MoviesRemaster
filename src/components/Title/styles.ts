import styled,{ css } from 'styled-components'
import media from 'styled-media-query'

  type props = {
    size:'large'|'medium'
  }

const Modifiers = {
    large:()=>css`
        font-size:70px;
        color: white;
        word-break: break-word;
        span{
            font-size:20px;
            color: lightgray;
            }
        ${media.lessThan('small')`
            font-size:50px;
            padding-left:10px;
        `}
    `,
    medium:()=>css`
        color: white;
        width: 250px;
        font-size:50px;
        margin-bottom:30px;
    `
}



export const Wrapper = styled.h1<props>`
${({size})=>css`
font-family:monospace;
${!!size && Modifiers[size]()}
`}

`