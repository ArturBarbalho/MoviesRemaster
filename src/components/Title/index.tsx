import * as S from './styles'

type props = {
  children:React.ReactNode
  size:'large'|'medium'
}

const Title = ({children, size}:props) => (
  <S.Wrapper size={size}> {children} </S.Wrapper>
)

export default Title