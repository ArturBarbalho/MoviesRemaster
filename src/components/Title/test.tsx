import { render, screen } from '@testing-library/react'

import Title from '.'

describe('<Title />', () => {
  it('should render the larger title ', () => {
    const { container } = render(<Title size='large' >DOU KYUU SEI</Title>)

    expect(screen.getByRole('heading', { name: /DOU KYUU SEI/ })).toBeInTheDocument()

    expect(screen.getByRole('heading', { name: /DOU KYUU SEI/})).toHaveStyle(
      'font-size:70px;'
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the medium title', () => {
    const { container } = render(<Title size='medium' >POPULAR</Title>)

    expect(screen.getByRole('heading', { name: /POPULAR/ })).toBeInTheDocument()

    expect(screen.getByRole('heading', { name: /POPULAR/ })).toHaveStyle(
      'width: 250px;'
    )

    expect(container.firstChild).toMatchSnapshot()
  
  })
})