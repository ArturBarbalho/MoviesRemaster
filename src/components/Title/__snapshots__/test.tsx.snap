// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render the larger title  1`] = `
<h1
  class="styles__Wrapper-sc-1buurg8-0 kNhFRm"
>
   
  DOU KYUU SEI
   
</h1>
`;

exports[`<Title /> should render the medium title 1`] = `
<h1
  class="styles__Wrapper-sc-1buurg8-0 fVOjsO"
>
   
  POPULAR
   
</h1>
`;
